tests/integration/plugins/test_plugins.py

```diff
@@ -521,3 +539,855 @@ async def test_knn_training_search(ops_test: OpsTest, deploy_type: str) -> None:
+    await http_request(ops_test, "DELETE", f"{dash_base_url}/api/sampple_data/{sample_data}")
+
+    # remove dashboards application
+    await ops_test.model.remove_application(DASHBOARDS_APP_NAME, block_until_done=True)
+
+
+@pytest.mark.parametrize("deploy_type", SMALL_DEPLOYMENTS)
+@pytest.mark.abort_on_fail
+async def test_sql_plugin(ops_test: OpsTest, deploy_type: str) -> None:
+    """Test that the SQL plugin is enabled and functional."""
+    leader_unit_ip = await get_leader_unit_ip(ops_test)
+    base_url = f"https://{leader_unit_ip}:9200"
+
+    # create index
+    await create_index(ops_test, APP_NAME, leader_unit_ip, TEST_INDEX)
+
+    # insert test docs
+    await bulk_insert(ops_test, APP_NAME, leader_unit_ip, bulk_encode(TEST_DOCS, TEST_INDEX))
+    await http_request(ops_test, "POST", f"{base_url}/{TEST_INDEX}/_refresh")
+
+    # select target doc
+    target = TEST_DOCS[-1]
+    target_id = target["id"]
+    target_text = target["passage_text"]
+
+    # create query
+    query = {"query": f"SELECT id, passage_text FROM {TEST_INDEX} WHERE id = '{target_id}'"}
+    endpoint = f"https://{leader_unit_ip}:9200/_plugins/_sql"
+    response = await http_request(ops_test, "POST", endpoint, query)
+    logger.info(f"SQL query response: {response}")
+    assert response.get("size") == 1, "Unexpected SQL result"
+    assert response.get("datarows")[0][-1] == target_text, "Unexpected SQL result"
+
+
+@pytest.mark.parametrize("deploy_type", SMALL_DEPLOYMENTS)
+@pytest.mark.abort_on_fail
+async def test_ism_and_job_scheduler_plugins(ops_test: OpsTest, deploy_type: str) -> None:
+    """Test that the ISM and job scheduler plugins are enabled and functional."""
+    leader_unit_ip = await get_leader_unit_ip(ops_test)
+    base_url = f"https://{leader_unit_ip}:9200"
+
+    # set job interval to 1m (min value)
+    settings = {
+        "persistent": {
+            "plugins.index_state_management.job_interval": 1,
+            "plugins.index_state_management.jitter": 0,
+        }
+    }
+
+    await http_request(
+        ops_test,
+        "PUT",
+        f"{base_url}/_cluster/settings",
+        settings,
+    )
+
+    # create index with alias
+    index_alias = "ism-test"
+    initial_index = f"{index_alias}-000001"
+    expected_end_index = (
+        f"{index_alias}-000002"  # after rollover the new index will have the number incremented
+    )
+    await create_index(
+        ops_test,
+        APP_NAME,
+        leader_unit_ip,
+        initial_index,
+        extra_index_settings={"plugins.index_state_management.rollover_alias": index_alias},
+    )
+
+    # set alias
+    await http_request(
+        ops_test,
+        "PUT",
+        f"{base_url}/{initial_index}/_alias/{index_alias}",
+        {"is_write_index": True},
+    )
+
+    # create policy to rollover index after min doc count (1)
+    policy_id = "rollover"
+    rollover = {
+        "policy": {
+            "description": "rollover",
+            "default_state": "hot",
```

Comment:
nice, and this is one of the use cases of large deployments, so that documents can be moved to different tiers (juju apps)